server/plugin.js

```diff
@@ -0,0 +1,70 @@
+/*
+ *   Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+import { first } from 'rxjs/operators';
+import { createAlertingCluster, createAlertingADCluster } from './clusters';
+import {
+  AlertService,
+  DestinationsService,
+  ElasticsearchService,
+  MonitorService,
+  AnomalyDetectorService,
+} from './services';
+import { alerts, destinations, elasticsearch, monitors, detectors } from '../server/routes';
+
+export class AlertingPlugin {
+  constructor(initializerContext) {
+    this.logger = initializerContext.logger.get();
+    this.globalConfig$ = initializerContext.config.legacy.globalConfig$;
+  }
+
+  async setup(core) {
+    // Get the global configuration settings of the cluster
+    const globalConfig = await this.globalConfig$.pipe(first()).toPromise();
+
+    // Create clusters
+    const alertingESClient = createAlertingCluster(core, globalConfig);
+    const adESClient = createAlertingADCluster(core, globalConfig);
```

Comment:
If you remove `createAlertingADCluster`, you could combine these into one client as well. I guess this is more of a choice on how you want to modularize them. It seems fine the way it is too. What do you think?

Reply:
I'm fine with combining the two files, if it works well. As I said, I didn't think much but try to keep as it is.😄 Yaliang added the file https://github.com/opendistro-for-elasticsearch/alerting-kibana-plugin/blame/master/server/clusters/alerting/createAlertingADCluster.js in the initial release of Anomaly Detection plugin.

Reply:
Well, I will ask his idea, and I found he didn't separate the 2 plugins in 2 clusters in AD's repo. 😅

Reply:
Got an idea from Yaliang. The advantage of separating the two files is that the two plugins can be modified separately, IMO, like the Header configuration. (Though AD didn't separate the 2 cluster clients ... ) This part is written by Mihir originally, I think I will remain it for now.

Reply:
Sure, makes sense, sounds good